### PR TITLE
Upgrading Dockerfile to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by zosorock**
Additionally:
- adding Docker Compose file with environmental variable support
- upgrading CircleCI executor images and sccache version
- upgrading Rust to 1.63
- upgrading sccache to 0.3.0

### CODE REVIEW

**Review by rpyell-aleo — APPROVED** [review]
I know the build is failing but it doesn't look like it has anything to do with these changes so LGTM.

**Comment by howardwu** [comment]
Let's queue this PR up for after #1871 is approved and merged

**Review by howardwu — APPROVED** [review]

**Pull request merged**
